src/Umbraco.Web.Common/Builder/UmbracoBuilderExtensions.cs

```diff
@@ -87,45 +87,17 @@ public static IUmbracoBuilder AddUmbracoCore(this IUmbracoBuilder builder)
                 options.Cookie.Name = "UMB_SESSION";
                 options.Cookie.HttpOnly = true;
             });
-
-            var syntaxProviders = new List<ISqlSyntaxProvider>();
-            var insertProviders = new List<IBulkSqlInsertProvider>();
-            var databaseCreators = new List<IEmbeddedDatabaseCreator>();
-
+            
             // Add supported databases
-            builder.Services.AddUmbracoSqlCeSupport(syntaxProviders, insertProviders, databaseCreators);
-            builder.Services.AddUmbracoSqlServerSupport(syntaxProviders, insertProviders, databaseCreators);
-
-            builder.Services.AddUnique<IAppPolicyCache>(factory => factory.GetRequiredService<AppCaches>().RuntimeCache);
-            builder.Services.AddUnique<IRequestCache>(factory => factory.GetRequiredService<AppCaches>().RequestCache);
+            builder.Services.AddUmbracoSqlCeSupport();
+            builder.Services.AddUmbracoSqlServerSupport();
 
-            var dbProviderFactoryCreator = new DbProviderFactoryCreator(
+            builder.Services.AddSingleton<IDbProviderFactoryCreator>(factory => new DbProviderFactoryCreator(
                 DbProviderFactories.GetFactory,
-                syntaxProviders,
-                insertProviders,
-                databaseCreators);
-
-            builder.Services.AddSingleton<IDbProviderFactoryCreator>(dbProviderFactoryCreator);
-
-            builder.Services.AddUnique<IIOHelper, IOHelper>();
-
-            var profiler = UmbracoCoreServiceCollectionExtensions.GetWebProfiler(builder.Config);
-            builder.Services.AddUnique<IProfiler>(profiler);
-
-            builder.Services.AddUnique<IProfilingLogger, ProfilingLogger>();
-            builder.Services.AddUnique<TypeLoader>(builder.TypeLoader);
-            builder.Services.AddUnique<IBackOfficeInfo, AspNetCoreBackOfficeInfo>();
-
-            builder.Services.AddUnique<IUmbracoDatabaseFactory, UmbracoDatabaseFactory>();
-            builder.Services.AddUnique<ISqlContext>(factory => factory.GetRequiredService<IUmbracoDatabaseFactory>().SqlContext);
-            builder.Services.AddUnique<IBulkSqlInsertProvider>(factory => factory.GetRequiredService<IUmbracoDatabaseFactory>().BulkSqlInsertProvider);
```

Comment:
^^ This line would cause a dependency cycle.

factory method for IDbProviderFactoryCreator resolves all IBulkSqlInsertProvider instances, and this line adds an extra one which resolves from IDbProviderFactoryCreator.

The only class which has a ctor dependency on IBulkSqlInsertProvider is UmbracoDatabase, but that instance is created via UmbracoDatabaseFactory